src/CustomMarshalers/src/System/Runtime/InteropServices/CustomMarshalers/EnumVariantViewOfEnumerator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace System.Runtime.InteropServices.CustomMarshalers
+{
+    internal class EnumVariantViewOfEnumerator : IEnumVARIANT, ICustomAdapter
+    {
+        public EnumVariantViewOfEnumerator(IEnumerator enumerator)
+        {
+            if (enumerator is null)
+            {
+                throw new ArgumentNullException(nameof(enumerator));
+            }
+
+            Enumerator = enumerator;
+        }
+        internal IEnumerator Enumerator { get; }
+
+        public IEnumVARIANT Clone()
+        {
+            if (Enumerator is ICloneable clonable)
+            {
+                return new EnumVariantViewOfEnumerator(clonable.Clone() as IEnumerator);
+            }
+            else
+            {
+                throw new COMException(SR.Arg_EnumNotCloneable, HResults.E_FAIL);
+            }
+        }
+
+        public int Next(int celt, object[] rgVar, IntPtr pceltFetched)
+        {
+            int numElements = 0;
+
+            try
+            {
+                if (celt > 0 && rgVar == null)
+                {
+                    return HResults.E_INVALIDARG;
+                }
+
+                while ((numElements < celt) && Enumerator.MoveNext())
+                {
+                    rgVar[numElements++] = Enumerator.Current;
+                }
+
+                if (pceltFetched != IntPtr.Zero)
+                {
+                    Marshal.WriteInt32(pceltFetched, numElements);
+                }
+            }
+            catch (Exception e)
+            {
+                return e.HResult;
+            }
+
+            return numElements == celt ? HResults.S_OK : HResults.S_FALSE;
+        }
+
+        public int Reset()
+        {
+            try
+            {
+                Enumerator.Reset();
+            }
+            catch (Exception e)
+            {
+                return e.HResult;
+            }
+
+            return HResults.S_OK;
+        }
+
+        public int Skip(int celt)
```

Comment:
Question: What's celt? I don't know much about this space.

Reply:
`celt` is the name of the parameter in the native API that we're implementing here. It stands for "count of elements" IIRC. I felt we should match the native API surface (as well as the interface declaration).